controller/controller.go

```diff
@@ -139,6 +141,8 @@ func (c *Controller) RunOnce(ctx context.Context) error {
 	}
 	sourceEndpointsTotal.Set(float64(len(endpoints)))
 
+	c.Provider.AdjustEndpoints(endpoints)
```

Comment:
Maybe AdjustEndpoints could return endpoint, so some could be removed or added by it on top of some being changed

Reply:
Oh that makes sense, good point! I'll make the change.

Reply:
I think actually that the signature is fine as it is: we are giving the provider as much freedom as needed on the array, so they can change it as needed, removing and adding, no need to return the endpoints.

Reply:
Slices in Go are not mutable, so no adding and removing is possible without returning new array

Reply:
Oh right 🤦